base_layer/core/src/transactions/transaction_components/transaction_output.rs

```diff
@@ -476,3 +481,98 @@ impl ConsensusDecoding for TransactionOutput {
         Ok(output)
     }
 }
+
+/// Performs batched range proof verification for an arbitrary number of outputs. Batched range proof verification gains
+/// above batch sizes of 2^8 = 256 gives diminishing returns, see <https://github.com/tari-project/bulletproofs-plus>,
+/// so the batch sizes are limited to 2^8.
+pub fn batch_verify_range_proofs(
+    prover: &RangeProofService,
+    outputs: &[&TransactionOutput],
+) -> Result<(), RangeProofError> {
+    // We need optimized power of two chunks, for example if we have 15 outputs, then we need chunks of 8, 4, 2, 1.
+    let power_of_two_vec = power_of_two_chunk_sizes(outputs.len(), 8);
+    debug!(
+        target: LOG_TARGET,
+        "Queueing range proof batch verify output(s): {:?}", &power_of_two_vec
+    );
+    let mut index = 0;
+    for power_of_two in power_of_two_vec {
+        let mut statements = Vec::with_capacity(power_of_two);
+        let mut proofs = Vec::with_capacity(power_of_two);
+        for output in outputs.iter().skip(index).take(power_of_two) {
+            statements.push(RistrettoAggregatedPublicStatement {
+                statements: vec![Statement {
+                    commitment: output.commitment.clone(),
+                    minimum_value_promise: 0,
+                }],
+            });
+            proofs.push(output.proof.to_vec().clone());
+        }
+        index += power_of_two;
+        prover.verify_batch(proofs.iter().collect(), statements.iter().collect())?;
+    }
+    Ok(())
+}
```

Comment:
This function should be in the bulletproofs+ crate. Think about it - Verifying batches of non power-of-two sets is something you'd be writing over and over if its not there.

Reply:
Yes it could move there and be totally transparent.

Reply:
I created an issue for it